lib/kcerts/certs.go

```diff
@@ -201,3 +202,12 @@ func NewOptions(mods ...Modifier) (*certOptions, error) {
 	}
 	return co, nil
 }
+
+// SSHCertTTL returns the remaining ttl of a cert, either when compared to current time, or it's total expiry.
+func SSHCertTTL(cert *ssh.Certificate, useCurrentTime bool) time.Duration {
```

Comment:
this function was very confusing for me to read :/ two notes:
- you assign `before` from Valid*After*
- and then subtract `before` from Valid*Before*

1) use different names for the variables?

2) Also, there's really two TTLs:
- total length of certificate life
- how longer the certificate has left to live.

Rather than a flag useCurrentTime, I would personally have two functions. Eg:

```
func SSHCertTotalLife(cert *ssh.Certificate) {
}

func SSHCertRemainingLife(cert *ssh.Certificate) {
}
```
3) Looking at: https://cs.opensource.google/go/x/crypto/+/a769d52b:ssh/certs.go;l=404

The ValidAfter and ValidBefore could also be negative. Given impact of computing those values and TTLs incorrectly, can you add more defense in depth?

Eg, what if ValidBefore is 0 or negative? What if ValidAfter is 0 or negative?

Reply:
I agree about the two different functions, implemented.

I would add the extra layer of defense in agent.AddCertificate (since detecting TTL is fundamentally an error less function).

Refusing if it's negative is ok, adding it in.